src/Widgets/AppContextMenu.vala

```diff
@@ -96,30 +82,27 @@ public class Slingshot.AppContextMenu : Gtk.Menu {
             });
         }
 
-#if HAS_PLANK
-        if (plank_client != null && plank_client.is_connected) {
+        if (Environment.find_program_in_path ("io.elementary.dock") != null) {
```

Comment:
shouldn't the existence of the Dock.dbus (or the GLib.Settings, if going that route) already confirm that dock is installed?

Reply:
I was going off of precedent from the AppCenter interface. The difference would be that if the dock is somehow not loaded when the context menu is opened you'd have no menuitem instead of an insensitive menu item, but I don't think that's a big deal. This whole menu will have to be rewritten for GTK 4 anyways, so happy to change it haha